src/Simulation/Simulation.jl

```diff
@@ -813,55 +813,60 @@ function refine!(sim::Simulation{T}, ::Type{WeightingPotential}, contact_id::Int
 end
 
 """
-    compute_min_tick_distance(grid, T, CS)
+    compute_min_tick_distance(grid)
 
-Compute the minimum grid spacing (tick distance) for a simulation grid.
+Compute the minimum grid spacing (tick distance) for a grid.
+Tick distances are computed as a small fraction of the physical axis lengths and then clamped to a safe numerical range.
 
 # Arguments
-- `grid` : a grid object containing 3 axes, either Cartesian or Cylindrical.
-- `T` : numeric type.
-- `CS` : coordinate system type, either `Cartesian` or `Cylindrical`.
+- `grid` : A simulation grid (`CylindricalGrid` or `CartesianGrid3D`).
 
-# Return
-- Tuple of minimum spacing `(Δx, Δy, Δz)` for Cartesian, `(Δr, Δφ, Δz)` for Cylindrical.
+# Returns
+A tuple of minimum tick distances in internal units, matching the grid type:
+- Cylindrical: `(Δr, Δφ, Δz)`
+- Cartesian: `(Δx, Δy, Δz)`
 """
-function compute_min_tick_distance(grid, T, CS)
+function compute_min_tick_distance(grid::CylindricalGrid{T}) where {T}
     min_tick = T(1e-12)
     max_tick = T(1e-5)
     fraction = T(1e-3)
 
-    if CS == Cylindrical
-        r_axis, phi_axis, z_axis = grid.axes
+    r_axis, phi_axis, z_axis = grid.axes
 
-        r_vals = collect(r_axis)
-        z_vals = collect(z_axis)
+    r_min = T(r_axis.interval.left)
+    r_max = T(r_axis.interval.right)
+    r_len = width(r_axis.interval)
 
-        r_min, r_max = T(first(r_vals)), T(last(r_vals))
-        z_min, z_max = T(first(z_vals)), T(last(z_vals))
-        r_mid = (r_min + r_max)/2
-        safe_r_mid = max(abs(r_mid), T(1e-30))
+    z_min = T(z_axis.interval.left)
+    z_max = T(z_axis.interval.right)
+    z_len = width(z_axis.interval)
 
-        r_len = r_max - r_min
-        z_len = z_max - z_min
+    r_mid = (r_min + r_max)/2
```

Comment:
You can also rewrite this, such that you don't need to define `r_min`, `r_max`, `z_min` and `z_max` anymore.
```suggestion
    r_mid = mean(r_axis.interval)
```